document/introduction/index.rst

```diff
@@ -1,6 +1,54 @@
 Introduction
 ============
 
-.. todo::
+WebAssembly (abbreviated Wasm [#wasm]_) is a *safe, portable, low-level code format*
+designed for efficient execution and compact representation.
+Its main goal is to enable high performance applications on the Web, but it does not make any Web-specific assumptions or provide Web-specific features, so can be employed in other environments as well.
 
-   All there is to say
+WebAssembly is an open standard developed by a `W3C Community Group <https://www.w3.org/community/webassembly/>`_ that includes representatives of all major browser vendors.
+
+This document describes version |release| of the standard.
+It is intended that it will be superseded by new incremental releases with additional features in the future.
+
+
+Design Goals
+------------
+
+.. index:: design goals, portability
+
+The design goals of WebAssembly are the following:
+
+* Fast, safe, and portable *semantics*:
+
+  * **Fast**: executes with near native code performance, taking advantage of capabilities common to all contemporary hardware.
+
+  * **Safe**: code is validated and executes in a memory-safe, sandboxed environment preventing data corruption or security breaches.
+
+  * **Deterministic**: fully and precisely defines valid programs and their behavior in a way that is easy to reason about informally and formally.
+
+  * **Hardware-independent**: can run on all modern hardware and CPUs, desktop or mobile devices and embedded systems alike.
+
+  * **Language-independent**: does not privilege nor penalize any particular programming or object model.
+
+  * **Platform-independent**: can be embedded in browsers, run as a stand-alone VM, or integrated in other environments.
+
+  * **Open**: programs can interoperate with their environment in a simple and universal manner.
```

Comment:
I agree that "open" is a bit overloaded, but it's commonly used it in the sense of an "open architecture". Not sure "connectable" has the same connotation. Maybe we can find something better? (I absolutely want to avoid saying "interoperable" here.)